### ISSUE TITLE
Sway crashes with for_window [app_id] move scratchpad

### ISSUE THREAD

**Issue by sentriz** [swaywm/sway#6239]
- **Sway Version:** sway version 1.6-a6dc829e (Apr 27 2021, branch 'master')
- **Debug Log:** https://termbin.com/p4tm
- **Configuration File:**
```
for_window [app_id="foot"] move scratchpad
exec foot
```
- **Stack Trace:** https://termbin.com/rhwt
- **Description:**

Hi, when running sway with this minimal config above like `sway -d -c minimal 2>log`, sway exits/crashes. 

The same thing happens with my normal config - writing a for_window rule with app_id to move to scratchpad ([foot](https://codeberg.org/dnkl/foot) the terminal emulator given as an example, but it happens for other programs too)

```
00:00:00.230 [sway/tree/arrange.c:293] Arranging workspace '1' at 0.000000, 0.000000
 err: wayland.c:1091: failed to read events from the Wayland socket: Broken pipe
 err: wayland.c:1506: failed to flush wayland socket: Broken pipe
 err: wayland.c:1506: failed to flush wayland socket: Broken pipe
 err: wayland.c:1506: failed to flush wayland socket: Broken pipe
warn: terminal.c:1529: slave exited with signal 1 (Hangup)
fish: Job 1, 'sway -d -c a' terminated by signal SIGSEGV (Address boundary error)
```

Would love any help with this. Thanks!

**Comment by emersion** [comment]
cc @Xyene: maybe related to https://github.com/swaywm/sway/pull/6224?

**Comment by sentriz** [comment]
after doing a git bisect, it seems the offending commit is 80128d23ba9f0a6a92284b2c6077e304f35e7a76

**Comment by RPigott** [comment]
https://github.com/swaywm/sway/commit/80128d23ba9f0a6a92284b2c6077e304f35e7a76#r50141849

**Comment by MuhammedTaha02** [comment]
Hello there. This feature works fine for me. But I have a problem. I press right ctrl + dot key to open Scratchpad (Control_R+period). But if I press the period key first and then the ctrl key, the scratchpad opens. If I press the ctrl key first and then the period key, the scratchpad will not open. How can i fix this problem @emersion. Thanks.

**Comment by emersion** [comment]
Please ask questions on IRC, not on GitHub.

**Comment by emersion** [comment]
Fixed in https://github.com/swaywm/sway/pull/6240.

**Comment by MuhammedTaha02** [comment]
> Please ask questions on IRC, not on GitHub.

It's not a question to ask on IRC @emersion. The problem is that sway handles period (dot) then ctrl (Control_R).

**Comment by Andonome** [comment]
I'm getting this crash. Display fails. Error message matches the OP's.

Sway version: 0.3.2
OS: Void Linux.

@emersion - I'll go an ask on IRC, as you've suggested above, but if everyone does that, wouldn't we lose all the information, like the fact that someone had this problem, and where is seems to be?

**Comment by kennylevinsen** [comment]
This specific issue was fixed 2.5 years ago, so this not what you are experiencing. You are also definitely not running sway 0.3.2, as no such release exists.

If you have a crash on the latest upstream sway release - and not, say, SwayFX - ensure you follow the guidelines forcreating a new issue with appropriate logs and stack traces.

**Comment by Andonome** [comment]
> This specific issue was fixed 2.5 years ago, so this not what you are experiencing. You are also definitely not running sway 0.3.2, as no such release exists.
> 
> If you have a crash on the latest upstream sway release - and not, say, SwayFX - ensure you follow the guidelines forcreating a new issue with appropriate logs and stack traces.

My mistake - I forgot I switched over to swayfx. Sorry about that!
